src/client/opamAdminCommand.ml

```diff
@@ -1005,13 +1005,38 @@ let add_constraint_command =
         let deps =
           OpamFormula.map (function
               | (n,c as atom) ->
-                if n = name then Atom (n, (add_cstr nv n c))
+                if n = name then Atom (n, (add_cstr cstr nv n c))
                 else Atom atom)
             deps0
         in
-        if deps <> deps0 then
+        let depopts0 = OpamFile.OPAM.depopts opam in
+        let conflicts0 = OpamFile.OPAM.conflicts opam in
+        let contains =
+          OpamFormula.fold_left (fun contains (n,_) ->
+              contains || n = name) false
+        in
+        let conflicts =
+          if contains depopts0 then
+            let icstr =
+              match cstr with
+              | OpamFormula.Atom (Constraint (relop, FString v)) ->
+                OpamFormula.Atom (Constraint (OpamFormula.neg_relop relop, FString v))
+              | _ -> OpamFormula.Empty
+            in
+            if contains conflicts0 then
+              OpamFormula.map (function
+                  | (n,c as atom) ->
+                    if n = name then Atom (n, (add_cstr icstr nv n c))
```

Comment:
`add_cstr` creates an AND, here we would need to create an OR instead
(if we had `conflicts: foo < 3` and we add the constraint `foo < 4`, we then need to conflict with `foo (< 3 | >= 4)` — `(< 3 & >= 4)` would be the empty set and actually cancel the existing conflicts)